lib/govspeak/post_processor.rb

```diff
@@ -94,7 +94,9 @@ def self.extension(title, &block)
       document.css(":not(thead) tr td:first-child").map do |el|
         next unless el.content.match?(/^#($|\s.*$)/)
 
-        el.content = el.content.gsub(/^#($|\s)/, "") # Replace '# ' and '#', but not '#Word'.
+        # Replace '# ' and '#', but not '#Word'.
+        # This runs on the first child of the element to preserve any links
+        el.children.first.content = el.children.first.content.gsub(/^#($|\s)/, "")
```

Comment:
I'm not sure I understand this so I may be asking a nonsense question. This previous ran on the content of the cell, but is now running on the first child of the cell. Does it still do what it was supposed to do if the cell has no children?

Reply:
If the cell has content, that content is the child (children.first) and it works. 
If the cell has content that includes a link it has multiple children.
If the cell has no content it is picked up later and the TH is turned into a TD.

Reply:
Okay, so even text content e.g. `This is the header` is counted as a child and processed correctly? (the `children.first` you referred to)

Reply:
Yes, see the test I added and note the input and output data.

Reply:
`<td>I am <a href="">a link</a> in a sentence</td>` - this breaks down as 3 children - text child, link child, text child. But the content of each can be targeted, and confusingly (what caused the bug) targeting content on the cell collects the content of each.

Reply:
Thanks, that's clarified things.